### ISSUE TITLE
Different loss function values on test in single gpu mode and multi gpu mode

### ISSUE THREAD

**Issue by kslazarev** [catboost/catboost#494]
Problem: different loss function values on test dataset/pairs in single gpu mode and multi gpu mode

### Environment

Catboost version: git master / 1d86c55eb0e1a2f18773d9cd988402e3c9367157
Operating System: Ubuntu 16.04
GPU: 2 x 1080 TI

### Test Case

Dataset: https://yadi.sk/d/M8oGYEdzD-EWdQ

Single GPU CMD: 

`catboost fit -f features.tsv -t features.tsv --learn-pairs train_pairs.tsv --test-pairs valid_pairs.tsv --train-dir test_case/ --task-type GPU --boosting-type Plain --cd cd --loss-function PairLogitPairwise -w 0.3 -r 0 -i 1000 --devices 0
`

> 999:    learn: 0.4734163        test: 0.6136901 best: 0.6136126 (962)   total: 54.2s    remaining: 0us
> Shrink model to first 963 iterations.

Multi GPU CMD: 

`catboost fit -f features.tsv -t features.tsv --learn-pairs train_pairs.tsv --test-pairs valid_pairs.tsv --train-dir test_case/ --task-type GPU --boosting-type Plain --cd cd --loss-function PairLogitPairwise -w 0.3 -r 0 -i 1000 --devices 0:1
`

> 999:    learn: 0.4740147        test: 0.5341348 best: 0.5341348 (999)   total: 34.1s    remaining: 0us

**Comment by Noxoomo** [comment]
Hi!

There is the bug in score calculation for your dataset.  Fix will be on github soon (currently you need to provide at least one group per device for learn and test sets. You have 6 groups in datasets, 5 for learn and one for test)

**Comment by Noxoomo** [comment]
fix on github

**Comment by kslazarev** [comment]
Have checked, now correct.
